### DISCUSSION TITLE
Photos: Timeline shows wrong date

### DISCUSSION THREAD

**Discussion by mpweya** [ente-io/ente#2446] [Q&A]
I'm sure this is more of a user error than it is a bug. But I'm happy to create an issue in case that would be more appropriate!

I've copied a bunch of photos from another device on my phone for syncing. Sadly, now all the photos are shown as created on the day I copied them when I view the timeline in ente. Opening the EXIF data on the pictures shows the correct picture creation date. Is there any way to tell ente to use the EXIF image creation date instead of the system file creation date if available?

**Comment by mnvr** [comment]
On the web and desktop apps, there is a button to "Fix times". You can select the relevant photos, click this action:

<img width="301" alt="Screenshot 2024-07-13 at 15 22 16" src="https://github.com/user-attachments/assets/6a4ffca9-1d7e-4135-8493-44aa90ba2351">

and then you'll have the ability to update the time for these photos from the Exif data.

<img width="354" alt="Screenshot 2024-07-13 at 15 23 00" src="https://github.com/user-attachments/assets/28e2a071-02d7-45ac-9b1a-8ea11a2ff970">

**Reply by spaasis** [reply]
@mnvr Could you add an option to fix the time based on the file name? For example Signal strips date/location metadata from exif but the file name contains the datetime (signal-2024-08-17-21-58-10-982-1.jpg). I'm currently going through importing a bunch of Signal convo images and I have to use the "Custom time" fix here, which is tedious to say the least 😅

**Reply by mnvr** [reply]
If you upload these files by using the web or desktop app, it will already pick the date and time from the filename. However, thanks!, because from your comment I noticed that Signal has apparently changed the format of the filenames, so I've now added support for the newer format too (https://github.com/ente-io/ente/pull/3176). 

I understand your comment was about the date time fixer, but if you now try to import "signal-2024-08-17-21-58-10-982-1.jpg" via the web app (change is already deployed to web), it'll pick up the correct time from the filename. So it should save on the tedium :)

**Reply by spaasis** [reply]
Great! I used exiftool to manually set the EXIF-data and reuploaded all the files, and that worked as expected. Good to hear that it should be easier the next time around. The fixer could probably use the file name option anyway, since there's undoubtedly some edge cases lying around :)
